draft-ounsworth-cfrg-kem-combiners.mkd

```diff
@@ -200,25 +228,20 @@ A post-quantum / traditional hybrid key encapsulation mechanism (hybrid KEM) as
 
 Building a PQ/T hybrid KEM requires a secure function which combines the output of both component KEMs to form a single output. Several IETF protocols are adding PQ/T hybrid KEM mechanisms as part of their overall post-quantum migration strategies, examples include TLS 1.3 [I-D.ietf-tls-hybrid-design], IKEv2 [I-D.ietf-ipsecme-ikev2-multiple-ke], X.509; PKIX; CMS [I-D.ounsworth-pq-composite-kem], OpenPGP [I-D.wussler-openpgp-pqc], JOSE / COSE (CITE once Orie's drafts are up).
 
+The traditional algorithm may in fact be a key transport or key agreement scheme, but since simple transformations exist to turn both of those schemes into KEMs, this document assumes that all cryptograhpic algorithms satisfy the KEM interface described in {{sec-kem-defn}}.
+
 ## KEM-based AKE
 
 The need for a KEM-based authenticated key establishment arises, for example, when two communicating parties each have long-term KEM keys (for example in X.509 certificates), and wish to involve both KEM keys in deriving a mutually-authenticated shared secret. In particular this will arise for any protocol that needs to provide post-quantum replacements for static-static (Elliptic Curve) Diffie-Hellman mechanisms. Examples include a KEM replacement for CMP's DHBasedMac {{I-D.ietf-lamps-cmp-updates}}.
 
 
 # KEM Combiner construction {#sec-kem-combiner}
 
-<!--
 TODO: read and cite the ETSI doc "Quantum-safe Hybrid Key Exchanges"
 https://www.etsi.org/deliver/etsi_ts/103700_103799/103744/01.01.01_60/ts_103744v010101p.pdf
 
 TODO: as per https://www.enisa.europa.eu/publications/post-quantum-cryptography-integration-study section 4.2, might need to specify behaviour in light of KEMs with a non-zero failure probability.
 
-TODO: read and cite
-Federico Giacon, Felix Heuer, and Bertram Poettering. KEM combiners. In Michel
-Abdalla and Ricardo Dahab, editors, PKC 2018: 21st International Conference on Theory
-and Practice of Public Key Cryptography, Part I, volume 10769 of Lecture Notes in Computer Science, pages 190–218. Springer, Heidelberg, March 2018.
--->
-
 
 A KEM combiner is a function that takes in two or more shared secrets `ss_i` and returns a combined shared secret `ss`, where all values are byte arrays.
```

Comment:
```suggestion
A KEM combiner is a function that takes in two or more shared secrets `ss_i` and returns a combined shared secret `ss`.
```
I think at this point you are defining a KEM combiner in a more conceptual way, therefore I would suggest to remove the technicality "where all values are byte arrays" (which should be implicitely clear anyway, since you describe them as inputs to hash functions later on).